core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java

```diff
@@ -140,6 +140,12 @@ public void testCompilationCircuitBreaking() throws Exception {
             scriptService.checkCompilationLimit(); // should pass
         }
         expectThrows(CircuitBreakingException.class, () -> scriptService.checkCompilationLimit());
+        scriptService.setMaxCompilationsPerMinute(0);
+        expectThrows(CircuitBreakingException.class, () -> scriptService.checkCompilationLimit());
+        scriptService.setMaxCompilationsPerMinute(Integer.MAX_VALUE);
+        for (int i = 0; i < randomIntBetween(1000, 10000); i++) {
```

Comment:
You might want to save the value of the limit rather than do it every iteration?